src/firewall/core/fw_nm.py

```diff
@@ -47,8 +48,26 @@ def nm_get_client():
     @return NM.Client instance if import was successful, None otherwise
     """
     global _nm_client
+    global _nm_client_timeout
+
+    def _release():
+        """
+        Release the client to avoid excess memory usage when libnm pushes
+        irrelevant (to firewalld) updates.
+        """
+        global _nm_client
+        global _nm_client_timeout
+        _nm_client = None
+        _nm_client_timeout = None
+
     if not _nm_client:
         _nm_client = NM.Client.new(None)
+        _nm_client_timeout = GLib.timeout_add_seconds(5, _release)
+    elif _nm_client_timeout:
```

Comment:
I think it still holds that at this place `_nm_client_timeout` cannot be zero.

But of course, it doesn't hurt much.

Patch lgtm.

Reply:
Why? Wouldn't repeated calls to `nm_get_client()` hit this `else`?

Reply:
Erm, I see what you mean. I had already pushed an `s/elif/else/` before I saw you comment. ;)